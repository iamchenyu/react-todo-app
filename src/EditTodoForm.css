.EditTodoForm {
  margin: 10px auto 10px auto;
}

.EditTodoForm input {
  height: 25px;
  width: 40%;
}

.EditTodoForm button {
  margin-left: 20px;
  height: 30px;
  width: 100px;
  border: none;
  font-family: "Prompt";
  letter-spacing: 1px;
}
