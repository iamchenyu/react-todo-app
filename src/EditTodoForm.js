import React, { useState } from "react";
import "./EditTodoForm.css";

const EditTodoForm = ({ id, text, onEdit }) => {
  const [todo, setTodo] = useState(text);

  const todoChangeHandler = (e) => {
    setTodo(e.target.value);
  };

  const todoEditHandler = (e) => {
    e.preventDefault();
    onEdit(e.target.todo.id, e.target.todo.value);
  };

  return (
    <form
      onSubmit={todoEditHandler}
      className="EditTodoForm"
      data-testid={text}
    >
      <input
        id={id}
        type="text"
        name="todo"
        value={todo}
        onChange={todoChangeHandler}
      />
      <button>Edit</button>
    </form>
  );
};

export default EditTodoForm;
