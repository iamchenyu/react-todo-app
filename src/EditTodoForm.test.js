import { render } from "@testing-library/react";
import EditTodoForm from "./EditTodoForm";

it("renders without crashing", () => {
  render(<EditTodoForm />);
});

it("matches the snapshot", () => {
  const { asFragment } = render(<EditTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});
