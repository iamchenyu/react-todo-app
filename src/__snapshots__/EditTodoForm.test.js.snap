// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <form
    class="EditTodoForm"
  >
    <input
      name="todo"
      type="text"
      value=""
    />
    <button>
      Edit
    </button>
  </form>
</DocumentFragment>
`;
